bcolz/carray_ext.pyx

```diff
@@ -734,6 +734,19 @@ cdef class chunks(object):
             scomp = schunk.read(ctbytes)
         return scomp
 
+    def __iter__(self):
+       self._iter_count = 0
+       return self
+
+    def __next__(self):
```

Comment:
I made some tests and I have a very good feeling about iterblocks, I would like to know what @CarstVaartjes and @esc think about this results and the future of this PR.

Please give me notice if this discussion should be moved to the mailing list.

In the following tests (v4 & v5 using iterblocks) (v2 @esc's code) (v1 & v3 much slower).
```python 
In [1]: from bcolz import carray

In [2]: from bcolz.carray_ext import test_v1, test_v2, test_v3, test_v4, test_v5

In [3]: import numpy as np

In [4]: n = 100000 + 3333

In [5]: x = np.arange(0, n, 1)

In [6]: c = carray(x, dtype='int64', chunklen=10000)

In [7]: %timeit test_v1(c)
1 loops, best of 3: 2.98 s per loop

In [8]: %timeit test_v2(c)
1000 loops, best of 3: 375 µs per loop

In [9]: %timeit test_v3(c)
1 loops, best of 3: 2.27 s per loop

In [10]: %timeit test_v4(c)
100 loops, best of 3: 4.1 ms per loop

In [11]: %timeit test_v5(c)
1000 loops, best of 3: 438 µs per loop
```
Code: https://github.com/FrancescElies/bcolz/blob/bench_iter_carray/bcolz/carray_ext.pyx#L2594-L2692

Hopefully next week I will be able to explore this a bit more.